src/core/Akka/IO/Tcp.cs

```diff
@@ -808,18 +810,32 @@ public sealed class CommandFailed : Event
             /// TBD
             /// </summary>
             /// <param name="cmd">TBD</param>
-            public CommandFailed(Command cmd)
-            {
-                Cmd = cmd;
-            }
+            public CommandFailed(Command cmd) => Cmd = cmd;
 
             /// <summary>
             /// TBD
             /// </summary>
             public Command Cmd { get; }
 
-            public override string ToString() =>
-                $"CommandFailed({Cmd})";
+            /// <summary>
+            /// Optionally contains the cause why the command failed.
+            /// </summary>
+            public Option<Exception> Cause { get; private set; } = Option<Exception>.None;
+
+            /// <summary>
+            /// Creates a copy of this object with a new cause set.
+            /// </summary>
+            [InternalApi]
+            public CommandFailed WithCause(Exception cause)
+            {
+                // Needs to be added with a mutable property for compatibility reasons
+                return new CommandFailed(Cmd) { Cause = cause };
```

Comment:
Not sure if this holds true in C# or we could just add a second constructor instead.

Reply:
We could just add a constructor overload and that would work

Reply:
Although I think this is probably a fine way of doing it